torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -723,49 +724,63 @@ def __init__(
-                             "which outputs embeddings that will be feed "
-                             "to the linear layer.")
-
-        self.in_channels = in_channels
+        if col_to_text_model_cfg is None:
+            raise ValueError("Please manually specify the "
+                             "`col_to_text_model_cfg`, which outputs "
+                             "embeddings that will be fed into the linear "
+                             "layer.")
+        # TODO: Support non-dictionary col_to_model_cfg
+        assert isinstance(col_to_text_model_cfg, dict)
+
+        self.in_channels_dict = {
+            col_name: model_cfg.out_channels
+            for col_name, model_cfg in col_to_text_model_cfg.items()
+        }
 
         super().__init__(out_channels, stats_list, stype, post_module,
                          na_strategy)
-        self.model = model
+
+        self.model_dict = torch.nn.ModuleDict({
+            col_name: model_cfg.model
+            for col_name, model_cfg in col_to_text_model_cfg.items()
+        })
 
     def init_modules(self) -> None:
         super().init_modules()
-        num_cols = len(self.stats_list)
-        self.weight = Parameter(
-            torch.empty(num_cols, self.in_channels, self.out_channels))
-        self.bias = Parameter(torch.empty(num_cols, self.out_channels))
+        self.weight_dict = torch.nn.ParameterDict()
+        self.bias_dict = torch.nn.ParameterDict()
+        for col_name, in_channels in self.in_channels_dict.items():
+            self.weight_dict[col_name] = Parameter(
+                torch.empty(in_channels, self.out_channels))
+            self.bias_dict[col_name] = Parameter(torch.empty(
+                self.out_channels))
         self.reset_parameters()
 
     def reset_parameters(self) -> None:
         super().reset_parameters()
-        torch.nn.init.normal_(self.weight, std=0.01)
-        torch.nn.init.zeros_(self.bias)
+        for col_name in self.weight_dict:
+            torch.nn.init.normal_(self.weight_dict[col_name], std=0.01)
+            torch.nn.init.zeros_(self.bias_dict[col_name])
 
     def encode_forward(
         self,
-        feat: TensorData,
+        feat: dict[str, MultiNestedTensor],
         col_names: list[str] | None = None,
     ) -> Tensor:
-        x = self.model(feat)
-        # [batch_size, num_cols, in_channels] *
-        # [num_cols, in_channels, out_channels]
-        # -> [batch_size, num_cols, out_channels]
-        x_lin = torch.einsum("ijk,jkl->ijl", x, self.weight)
-        # [batch_size, num_cols, out_channels] + [num_cols, out_channels]
-        # -> [batch_size, num_cols, out_channels]
-        x = x_lin + self.bias
+        xs = []
+        for i, col_name in enumerate(col_names):
+            # [batch_size, 1, in_channels]
+            x = self.model_dict[col_name]({
+                key: feat[key][:, i]
+                for key in feat
+            })
```

Comment:
This operation assumes text input, right?

Reply:
I think this operation just selects each column right? Let's say if we have multiple columns for image data. I think image can also reuse this? For example, https://huggingface.co/docs/transformers/model_doc/vit#transformers.ViTModel.forward.example
```python
>>> from datasets import load_dataset
>>> dataset = load_dataset("huggingface/cats-image")
>>> image = dataset["test"]["image"][0]
>>> image_processor = AutoImageProcessor.from_pretrained("google/vit-base-patch16-224-in21k")
>>> inputs = image_processor(image, return_tensors="pt")
>>> inputs
{'pixel_values': tensor([[[[ 0.1137,  0.1686,  0.1843,  ..., -0.1922, -0.1843, -0.1843],
          [ 0.1373,  0.1686,  0.1843,  ..., -0.1922, -0.1922, -0.2078],
          [ 0.1137,  0.1529,  0.1608,  ..., -0.2314, -0.2235, -0.2157],
          ...,
          [ 0.8353,  0.7882,  0.7333,  ...,  0.7020,  0.6471,  0.6157],
          [ 0.8275,  0.7961,  0.7725,  ...,  0.5843,  0.4667,  0.3961],
          [ 0.8196,  0.7569,  0.7569,  ...,  0.0745, -0.0510, -0.1922]],

         [[-0.8039, -0.8118, -0.8118,  ..., -0.8902, -0.8902, -0.8980],
          [-0.7882, -0.7882, -0.7882,  ..., -0.8745, -0.8745, -0.8824],
          [-0.8118, -0.8039, -0.7882,  ..., -0.8902, -0.8902, -0.8902],
          ...,
          [-0.2706, -0.3176, -0.3647,  ..., -0.4275, -0.4588, -0.4824],
          [-0.2706, -0.2941, -0.3412,  ..., -0.4824, -0.5451, -0.5765],
          [-0.2784, -0.3412, -0.3490,  ..., -0.7333, -0.7804, -0.8353]],

         [[-0.5451, -0.4667, -0.4824,  ..., -0.7412, -0.6941, -0.7176],
          [-0.5529, -0.5137, -0.4902,  ..., -0.7412, -0.7098, -0.7412],
          [-0.5216, -0.4824, -0.4667,  ..., -0.7490, -0.7490, -0.7647],
          ...,
          [ 0.5686,  0.5529,  0.4510,  ...,  0.4431,  0.3882,  0.3255],
          [ 0.5451,  0.4902,  0.5137,  ...,  0.3020,  0.2078,  0.1294],
          [ 0.5686,  0.5608,  0.5137,  ..., -0.2000, -0.4275, -0.5294]]]])}
>>> inputs['pixel_values'].shape
torch.Size([1, 3, 224, 224])
```

Reply:
That makes sense. Addressed in https://github.com/pyg-team/pytorch-frame/pull/270#discussion_r1421172944